tests/test_dsp_filtering.py

```diff
@@ -0,0 +1,168 @@
+    np.array([1000, 2000]),
+    np.array([500,6700,15200,1200]),
+])
+@pytest.mark.parametrize("frac",[
+    1,3,
+])
+@pytest.mark.parametrize("n_sigs",[
+    1,2,
+])
+def test_band_filtering(freq,frac,n_sigs):
+    """Test freq band data estimation."""
+
+    ff = np.array([freq]*n_sigs)
+    scale = (np.ones_like(ff)*
+             np.arange(0.1,.5,ff.shape[1])*
+             np.arange(0.5,1,ff.shape[0]))
+    signal_split_freqs = pf.signals.sine(frequency=ff, n_samples=441)
+    signal_split_freqs.time = (scale[...,None]*signal_split_freqs.time)
+
+
+
+    signal_combined = pf.Signal(data=np.sum(signal_split_freqs.time, axis=1),
+                                sampling_rate=signal_split_freqs.sampling_rate)
+
+    band_sig,_ = sp.dsp.band_filter_signal(signal=signal_combined,
+                                         frequencies=freq,
+                                         num_fractions=frac,
+                                         )
+
+    assert band_sig.cshape==(signal_split_freqs.cshape)
+
+    npt.assert_allclose(np.argmax(np.abs(band_sig.freq),axis=-1),
+                        np.argmax(np.abs(signal_split_freqs.freq),axis=-1))
+
+
+def test_band_filtering_inputs():
+    frequencies = np.array([0,1,2,3])
+    with pytest.raises(
+            ValueError,
+            match="Input frequencies must be greater than zero."):
+        sp.dsp.band_filter_signal(frequencies=frequencies,
+                                signal=pf.signals.noise(n_samples=400),
+                                num_fractions=1)
+
+
+    frequencies = np.array([10,20000,-3])
+    with pytest.raises(
+            ValueError,
+            match="Input frequencies must be greater than zero."):
+        sp.dsp.band_filter_signal(frequencies=frequencies,
+                                signal=pf.signals.noise(n_samples=400),
+                                num_fractions=1)
+
+
+    with pytest.raises(
+            ValueError,
+            match="Number of octave fractions must be greater than zero."):
+        sp.dsp.band_filter_signal(frequencies=np.array([1,2,3]),
+                                signal=pf.signals.noise(n_samples=400),
+                                num_fractions=0)
+
+    with pytest.raises(
+            ValueError,
+            match="Number of octave fractions must be greater than zero."):
+        sp.dsp.band_filter_signal(frequencies=np.array([1,2,3]),
+                            signal=pf.signals.noise(n_samples=400),
+                            num_fractions=-5)
+
+
+def test_closest_frac_octave_data_bandwidth():
+    num_fractions = 3
+    frequency_range = (25, 12e3)
+    _, fband_centers, cutoffs = pf.dsp.filter.fractional_octave_frequencies(
+        num_fractions=num_fractions,
+        frequency_range=frequency_range,
+        return_cutoff=True,
+        )
+    bw, _ = sp.dsp._closest_frac_octave_data(
+        frequencies=fband_centers, num_fractions=num_fractions)
+    npt.assert_allclose(bw, cutoffs[1]-cutoffs[0])
```

Comment:
```suggestion
    bw, index = sp.dsp._closest_frac_octave_data(
        frequencies=fband_centers[index_expected], num_fractions=num_fractions)
    npt.assert_allclose(bw, cutoffs[1]-cutoffs[0])
    npt.assert_allclose(fband_centers[index_expected], fband_centers[index], )
```

nice, here we vould also test the index parameter,